homeassistant/components/conversation/agent_manager.py

```diff
@@ -109,7 +110,19 @@ async def async_converse(
                 dataclasses.asdict(conversation_input),
             )
         )
-        result = await method(conversation_input)
+        try:
+            result = await method(conversation_input)
+        except HomeAssistantError as err:
+            intent_response = intent.IntentResponse(language=language)
+            intent_response.async_set_error(
+                intent.IntentResponseErrorCode.UNKNOWN,
+                str(err),
+            )
+            return ConversationResult(
```

Comment:
Let's make sure it's still tracked in the trace:
```suggestion
            result = ConversationResult(
```

Reply:
Fixed, and added a test verifying it shows up in the trace.